docs/ref/models/querysets.txt

```diff
@@ -2399,6 +2399,47 @@ Class method that returns an instance of :class:`~django.db.models.Manager`
 with a copy of the ``QuerySet``’s methods. See
 :ref:`create-manager-with-queryset-methods` for more details.
 
+
+``explain()``
+~~~~~~~~~~~~~
+
+.. versionadded: 2.1
+
+.. method:: explain(verbose=False, format=None)
+
+Returns the execution plan of the current queryset as a string. This details
+how the database would execute the given query, highlighting any indexes or
+joins used. For example, when using ``postgresql``::
+
+    >>> print(Blog.objects.filter(title="My Blog").explain())
+    Seq Scan on blog  (cost=0.00..35.50 rows=10 width=12)
+      Filter: (title = 'My Blog'::bpchar)
+
+Some databases (like MySQL and Postgres) accept flags that can return more
+information about the query. These flags can be passed as case-insensitive keyword
+arguments to the function. For example with Postgres::
+
+    >>> print(Blog.objects.filter(title="My Blog").explain(verbose=True))
+    Seq Scan on public.blog  (cost=0.00..35.50 rows=10 width=12) (actual time=0.004..0.004 rows=10 loops=1)
+      Output: id, title
+      Filter: (blog.title = 'My Blog'::bpchar)
+    Planning time: 0.064 ms
+    Execution time: 0.058 ms
+
+.. warning::
+
+    On some supported databases these flags can cause the query to be executed
```

Comment:
Perhaps worth explicitly mentioning `analyze=True` for PostgreSQL?

Reply:
It could be good to include, I'm not sure on the wording though. Do you have any ideas?

Reply:
I'm not sure what "supported databases" means. Is "supported" adding some meaning?